lib/puppet/util/package/version/pip.rb

```diff
@@ -0,0 +1,170 @@
+module Puppet::Util::Package::Version
+  class Pip
+    include Comparable
+
+    def self.parse(version)
+      raise ValidationFailure.new(version.to_s) unless version.is_a? String
```

Comment:
we should be consistent when raising errors, see line: 122
i suggest using `raise(ErrorClass, message)` for better readability.